packages/task/src/browser/task-configurations.ts

```diff
@@ -0,0 +1,189 @@
+    /** task configuration file name */
+    protected readonly TASKFILE = 'tasks.json';
+
+    protected client: TaskConfigurationClient | undefined = undefined;
+
+    constructor(
+        @inject(ILogger) protected readonly logger: ILogger,
+        @inject(FileSystemWatcherServer) protected readonly watcherServer: FileSystemWatcherServer,
+        @inject(FileSystem) protected readonly fileSystem: FileSystem
+    ) {
+        this.toDispose.push(watcherServer);
+
+        watcherServer.setClient({
+            onDidFilesChanged: async changes => {
+                try {
+                    await this.onDidTaskFileChange(changes);
+                    if (this.client) {
+                        this.client.taskConfigurationChanged(this.getTaskLabels());
+                    }
+                } catch (err) {
+                    this.logger.error(err);
+                }
+            }
+        });
+
+        this.toDispose.push(Disposable.create(() => {
+            this.tasksMap.clear();
+            this.client = undefined;
+        }));
+    }
+
+    setClient(client: TaskConfigurationClient) {
+        this.client = client;
+    }
+
+    dispose() {
+        this.toDispose.dispose();
+    }
+
+    /** watches for a task configuration file under the given root URI  */
+    async watchConfigurationFile(rootUri: string): Promise<void> {
+        let configFile: string | undefined;
+
+        configFile = await this.getConfigFile(rootUri);
+        if (configFile) {
+            if (this.watchedConfigFileUri !== configFile) {
+                this.watchedConfigFileUri = configFile;
+                const watchId = await this.watcherServer.watchFileChanges(configFile);
+                this.toDispose.push(Disposable.create(() =>
+                    this.watcherServer.unwatchFileChanges(watchId))
+                );
+
+                const tasks = await this.readTasks(configFile);
+                if (tasks) {
+                    for (const task of tasks) {
+                        this.tasksMap.set(task.label, task);
+                    }
+                }
+            }
+        } else {
+            this.logger.info(`Config file tasks.json does not exist under ${rootUri}`);
+        }
+    }
+
+    /** returns the list of known task labels */
+    getTaskLabels(): string[] {
+        return [...this.tasksMap.keys()];
+    }
+
+    /** returns the task configuration for a given label */
+    getTask(taskLabel: string): TaskOptions | undefined {
+        if (this.tasksMap.has(taskLabel)) {
+            return this.tasksMap.get(taskLabel);
+        } else {
+            return undefined;
+        }
+    }
+
+    protected async getConfigFile(rootDir: string): Promise<string | undefined> {
+        // compute URI of the tasks file to watch
+        const cfgFile = new URI(rootDir).resolve(this.TASKFILEPATH).resolve(this.TASKFILE).toString();
+
+        if (await this.fileSystem.exists(cfgFile)) {
+            return cfgFile;
+        }
```

Comment:
Missing `else { return Promise.resolve(undefined) } `

Reply:
Following previous changes, this is no longer applicable (method no longer async now that it only computes the potential task config file URI)